unittests/cpp/common/tests/Device_test.cpp

```diff
@@ -276,6 +283,87 @@ TEST_F(DeviceTest, Device_Create) {
 
 // --- TryMultiSetValue Tests ---
 
+// add expect calls
+// child param is not mock
+
+// 1.0: Error Case - Fails to write to a child when parent struct ("product") is read-only
+TEST_F(DeviceTest, TryMultiSetValue_ReadOnlyStruct) {
+auto parentParam = std::make_shared<MockParam>();
+    auto parentInfo  = std::make_shared<MockParamDescriptor>();
+    auto& parentDesc = *parentInfo;
+
+    setupMockParam(*parentParam, "/product", parentDesc, /*isReadonly*/false, /*access*/0, adminScope_);
```

Comment:
```suggestion
    setupMockParam(*parentParam, "/product", parentDesc, /*isArray*/false, /*size*/0, adminScope_);
```
```
/**
 * @brief Helper function to set up common mock parameter expectations
 * @param param The mock parameter to set up
 * @param oid The OID to return
 * @param descriptor The descriptor to return
 * @param isArray Whether this is an array type (defaults to false)
 * @param size The array size if it's an array type (defaults to 0)
 * @param scope The scope to return (defaults to monitor)
 */
inline void setupMockParam
```
4th param is `isArray` not `isReadonly`
5th param is `size` not `access`